src/H5Gint.c

```diff
@@ -747,7 +747,8 @@ H5G_nameof(const H5G_t *grp)
     /* Use FUNC_ENTER_NOAPI_NOINIT_NOERR here to avoid performance issues */
     FUNC_ENTER_NOAPI_NOINIT_NOERR
 
-    FUNC_LEAVE_NOAPI(grp ? &(grp->path) : NULL)
+    /* TODO: should not cast away const! */
+    FUNC_LEAVE_NOAPI(grp ? (H5G_name_t *)&(grp->path) : NULL)
```

Comment:
Please don't cast away const, it's hiding the underlying design issue, which should be fixed.